### DISCUSSION TITLE
collect multiple meters

### DISCUSSION THREAD

**Discussion by gemu2015** [jomjol/AI-on-the-edge-device#629] [Show and tell]
danke für dieses tolle Projekt! Nachdem ich lange Zeit erfolglos versucht habe meinen Gaszähler per induktivem Sensor abzutasten war das die Lösung

Damit habe ich jetzt alle meine Zähler mit 2 ESP32 erfasst.
vielleicht ist meine Lösung auch für andere interessant.
Ein ESP32 sammelt über 5 serielle Schnittstellen und einen Counter insgesamt 7 Meter Daten und speichert sie auf SD Karte.
Die CAM Software habe ich so modifiziert dass sie den Meterwert als OBIS asci über serielle Schnittstelle ausgibt. Damit habe ich auch bei Ausfall des WLANS keinerlei Datenverlust.

Insgesamt erfasse ich 7 Meter:
2 Modbus Stromzähler SDM530 u. SDM 630 für Solaranlagen und Wallbox
1 SML Zweirichtungszähler als Haupthauszähler
1 OBIS Stromzähler als Einspeisezähler
1 EBUS  für Heizung und Solarwärmedaten
1 Gaszähler über ESP32CAM
1 Wasserzähler über Hall-Sensor.

Wenn Interesse besteht kann ich gerne einen PR für die Modifikation serielle Übertragung machen.

Die Daten werden alle 15 Minuten und jeweils um 0 Uhr geloggt. Ausserdem wird noch eine separate Datei für jeden Tag angelegt. 

Übertragung aller Daten per MQTT ist natürlich weiterhin zusätzlich zur SD Karte möglich.

Die Logdateien werden jeden Monat per email versendet und können auch jederzeit ausgelesen werden

Das ganze läuft auf Tasmota mit Scriptsprache

![147753842-a666ce04-8246-4490-9ec3-8a5ccb59b78d](https://user-images.githubusercontent.com/11647075/152197383-58895bfa-bc2e-49cd-b372-a879c538d83a.jpg)

**Comment by littlebilly** [comment]
Hallo Gemu,
schön dich hier zu treffen. Das ist wirklich ein tolles Projekt. Du hast mir ja damals geholfen damit ich mit deinem Scripting den 
Zisterneninhalt berechnen kann. Nochmals vielen Dank dafür. 
Nachdem ich jetzt ja dank diesem Projekt meinen Wasserverbrauch litergenau erfassen kann habe ich mir überlegt für was ich das nützen könnte.
Idee: Ich hänge vor meinen Geschirrspüler einen Gosund mit Tasmota und Scripting geflasht. Sobald der Geschirrspüler einschaltet holt  sich Tasmota (Gosund) vom Wasserzähler (Digitizer - AI on the edge) den Zählerstand zu Beginn und am Ende des Waschprogramms (Verbrauch 0) den Zählerstand Ende. Mit Hilfe deines Scripting wird dann im Gosund der Wasserverbrauch berechnet und mit USE_SCRIPT_WEB_DISPLAY dargestellt.
Voraussetzung ist natürlich dass in diesem Zeitraum kein weiterer Wasserverbrauch erfolgt. (das lässt sich ja regeln)
Ich möchte nämlich schon lange mal testen welches Spülprogramm für unseren Zweck das geeignetste ist.
Das müsste sich doch realisieren lassen.
LG Billy

**Reply by gemu2015** [reply]
Hi Billy,

Nehme an du liest den Wasserzähler über MQTT aus. Dann wäre es fast einfacher auch den GOSUND über MQTT auszulesen und das alles auf dem Raspi zu machen.

Sonst müsstest du ja den Zählerstand vom MQTT Broker aus an den GOSUND schicken.
Wenn du das machen willst kannst du natürlich dann ganz einfach per Script den Wasserverbrauch bestimmen.

Wie gesagt ich lese den AI ESP32 per serielle Leitung aus und übertrage sämtliche interessanten Variablen zusätzlich zur Speicherung auf SD Karte über globale (Multicast UDP) Variablen im lokalen Netz.

Die Auswertung und Webgrafiken habe ich alle auf einem separaten ESP32 (m5stack core2) realisiert weil ich den Raspi komplett loswerden will.

LG
Gerhard

**Comment by DieterPN** [comment]
Hallo Gerhard,

an der Ausgabe über eine serielle Schnittstelle bin ich interessiert - kannst du deine Modifikation der CAM Software teilen?

Grüße

Dieter

**Reply by gemu2015** [reply]
Hallo Dieter,
hier die modifizierten Files basierend auf der aktuellen Version.
in obis.h kannst du einstellen welcher Pin für die serielle Ausgabe genommen werden soll.
bleibt ja eigentlich nur Pin 12 den ich genommen habe.

als OBIS code habe ich 1-0:1.8.0*255(%s)\r\n genommen. Kannst du jederzeit ändern.

Gruß
Gerhard

[AI_OTE.zip](https://github.com/jomjol/AI-on-the-edge-device/files/8056007/AI_OTE.zip)

PS. ein Versuch einen PR zu machen ist wegen einiger unerwarteter Probleme gescheitert. Deshalb etwas umständlich hier nur die Files.

**Comment by DieterPN** [comment]
Hallo Gerhard, Danke - werde ich nächstes Wochenende mal durchgehen. Grüße Dieter

**Comment by AuqW9cQ** [comment]
Hallo Gemu,

würde gerne zusätzlich zu meinem Wasserzähler via Cam meinen EMH Drehstromzähler auslesen, wenn möglich nur mit einem ESP.

Hab bis jetzt immer fertige Binaries verwendet, daher bin ich damit etwas überfordert. Wäre nett wenn du mir da etwas zur Verfügung stellen könntest.

**Reply by gemu2015** [reply]
Hi,
Da würde ich an deiner Stelle zuerst mal den EMH Zähler auslesen. Hier gibt es hilfreiche Seiten dazu:

https://ottelo.jimdofree.com/stromzähler-auslesen-tasmota/

https://forum.creationx.de/forum/index.php?thread/1095-d0-zähler-sml-auslesen-mit-tasmota/

wenn du das geschafft hast melde dich noch mal.

Kann dir dann helfen wie du den Wasserzähler addieren kannst.

**Reply by AuqW9cQ** [reply]
Danke für deine Antwort,

sollte ich für den Wasserzähler nicht "AI on the Edge" flashen und darauf aufbauend dann den Stromzähler integrieren?

Hab zuerst mal Tasmota geflasht, aber das mit dem Scripting wollte so nicht klappen... im Moment ist AIotE geflasht, aber da weiß ich auch nicht wie ich da den IR-Lesekopf einbinden kann.

Bin ich so komplett auf dem Holzweg?

**Reply by gemu2015** [reply]
Den Stromzähler (über serielle Schnittstelle) in die Cam Firmware zu integrieren wäre ein ganz neues Projekt.

Was ich oben beschrieben hatte ist ein ESP32 der alle meine Verbrauchsdaten erfasst und eine zusätzliche ESP32 Cam die meinen Gaszähler über eine serielle Schnittstelle mit diesem ESP32 verbindet.
Dieser ESP32 speichert dann alle Daten (insgesamt 7 Messquellen) auf einer SD Karte.
Ich wollte den Raspberry mit Iobroker loswerden und alles nur mit ESPs machen inklusive Verbrauchsgrafiken.
hier sind auch einige Screenshots der Grafiken dabei: https://tasmota.github.io/docs/Scripting-Language/

Wenn du ohnehin eine zentralen MQTT Server hast würde ich einfach die CAM und den Zähler getrennt mit 2 ESPs über MQTT mit diesem verbinden.
